src/post/workshed_utils.ts

```diff
@@ -67,7 +67,30 @@ export function getBestCycle(): TrainSet.Cycle {
   return [...trainCycle];
 }
 
-export function offsetDefaultPieces(offset: number): TrainSet.Cycle {
+function valueStation(station: TrainSet.Station): number {
+  if (station === TrainSet.Station.COAL_HOPPER) {
+    return valueStation(getBestCycle()[1]);
+  }
+  return GOOD_TRAIN_STATIONS.find(({ piece }) => piece === station)?.value() ?? 0;
+}
+
+function valueOffset(offset: number): number {
+  const firstFortyTurns = 5 * sum(getBestCycle(), valueStation);
+  const extraTurns = sum(getBestCycle().slice(0, offset), valueStation);
+  return (firstFortyTurns + extraTurns) / (40 + offset);
+}
+
+let bestOffset: number | null = null;
+function getBestOffset(): number {
+  bestOffset ??= maxBy([2, 3, 4, 5, 6, 7], valueOffset);
+  return bestOffset;
+}
+
+export function getPrioritizedStations(): TrainSet.Station[] {
+  return getBestCycle().slice(0, getBestOffset());
+}
```

Comment:
There's an off-by-one error here I think.
When `bestOffset` is `n`, we consider up to `n-1` stations after the Coal Hopper (hence when `n` is 2, we consider the 1st station after). Here we only consider up to `n=7` in the `maxBy`, which is only 6 after the coal hopper, and we will never consider the last station